packages/block-editor/src/components/link-control/index.js

```diff
@@ -251,59 +369,121 @@ function LinkControl( {
+			directLinkEntryTypes.includes(
+				suggestions[ 0 ].type.toLowerCase()
+			);
+		const shouldShowCreateEntity =
+			createSuggestion &&
+			! isSingleDirectEntryResult &&
+			! isInitialSuggestions;
+
 		// According to guidelines aria-label should be added if the label
 		// itself is not visible.
 		// See: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/listbox_role
-		const ariaLabel = isInitialSuggestions ? undefined : labelText;
-		const SearchResultsLabel = (
+		const searchResultsLabelId = `block-editor-link-control-search-results-label-${ instanceId }`;
+		const labelText = isInitialSuggestions
+			? __( 'Recently updated' )
+			: sprintf( __( 'Search results for "%s"' ), inputValue );
+
+		const searchResultsLabel = (
 			<span
-				className="block-editor-link-control__search-results-label"
+				className={ classnames(
+					'block-editor-link-control__search-results-label',
+					{
+						'screen-reader-text': ! isInitialSuggestions,
+					}
+				) }
 				id={ searchResultsLabelId }
-				aria-label={ ariaLabel }
 			>
 				{ labelText }
 			</span>
 		);
 
 		return (
 			<div className="block-editor-link-control__search-results-wrapper">
-				{ isInitialSuggestions ? (
-					SearchResultsLabel
-				) : (
-					<VisuallyHidden>{ SearchResultsLabel }</VisuallyHidden>
-				) }
-
+				{ searchResultsLabel }
 				<div
 					{ ...suggestionsListProps }
 					className={ resultsListClasses }
 					aria-labelledby={ searchResultsLabelId }
 				>
-					{ suggestions.map( ( suggestion, index ) => (
-						<LinkControlSearchItem
-							key={ `${ suggestion.id }-${ suggestion.type }` }
-							itemProps={ buildSuggestionItemProps(
-								suggestion,
-								index
-							) }
-							suggestion={ suggestion }
-							onClick={ () => {
-								onChange( { ...value, ...suggestion } );
-								stopEditing();
-							} }
-							isSelected={ index === selectedSuggestion }
-							isURL={ manualLinkEntryTypes.includes(
-								suggestion.type.toLowerCase()
-							) }
-							searchTerm={ inputValue }
-						/>
-					) ) }
+					{ suggestions.map( ( suggestion, index ) => {
+						if (
+							shouldShowCreateEntity &&
+							CREATE_TYPE === suggestion.type
+						) {
+							return (
+								<LinkControlSearchCreate
+									searchTerm={ inputValue }
+									onClick={ async () => {
+										try {
+											cancelableOnCreate.current = makeCancelable(
+												handleOnCreate(
+													suggestion.title
+												)
+											);
+
+											await cancelableOnCreate.current
+												.promise;
+											// Only stop editing if not rejected
+											stopEditing();
+										} catch ( error ) {
+											if ( error && error.isCanceled ) {
+												return; // bail if canceled to avoid setting state
+											}
+
+											setErrorMessage(
+												__(
+													'An unknown error occurred during creation. Please try again.'
+												)
+											);
```

Comment:
Apologies if this is a silly question, but why is the error unknown? Should we be able to check to see if the `error` passed a message?

Reply:
I've added this check and left the fallback in place.